doc/rst/language/spec/ranges.rst

```diff
@@ -307,24 +310,29 @@ The type of a range literal is a range with the following parameters:
 -  ``boundedType`` is a value of the type ``BoundedRangeType`` that is
    determined as follows:
 
-   -  ``bounded``, if both the low bound and the high bound expressions
+   -  ``bounded``, if both the lower bound and the upper bound expressions
       are given,
 
-   -  ``boundedLow``, if only the high bound expression is given,
+   -  ``boundedLow``, if only the upper bound expression is given,
 
-   -  ``boundedHigh``, if only the low bound expression is given,
+   -  ``boundedHigh``, if only the lower bound expression is given,
 
    -  ``boundedNone``, if neither bound expression is given.
 
 -  ``stridable`` is ``false``.
 
 The value of a range literal is as follows:
 
--  The low bound is given by the low bound expression, if present, and
+-  The low bound is given by the lower bound expression, if present, and
    is -:math:`\infty` otherwise.
 
--  The high bound is given by the upper bound expression, if present,
-   and is +\ :math:`\infty` otherwise.
+- For ranges constructed with ``..``, the high bound is given by the
+  upper bound expression, if present, and is +\ :math:`\infty`
+  otherwise.
+
+- For ranges constructed with ``..<``, the high bound is one less than
+  the upper bound expression, if present, and is +\ :math:`\infty`
+  otherwise.
```

Comment:
This is not an issue that is introduced in this PR, but, I find the language a bit tautological here. "high bound is given by the upper bound expression" doesn't sound like it is giving any valuable new information because the "upper bound expression" is clear enough to me.  Reading the changes, I was expecting that the difference between `..` and `..<` would use the term "inclusive" or similar. So I'd find

> For ranges constructed with ``..``, the value represented with the upper bound expression is included in the range.

A more precise description.

Another alternative is just explicitly saying `..` is a closed-interval, `..<` is an open-interval.

Reply:
> I find the language a bit tautological here

I don't necessarily disagree, but was nervous about changing things too dramatically.  Specifically, I think the flow of this chapter is to introduce ranges as a semi-mathematical concept defined by low bound, high bound, stride, and alignment values.  Then this section talks specifically about how code of the form `b1..b2` translates into those four values.  So I agree it's a bit obvious, but I think of this as being the interface between the code and the abstract values described previously.  And back to my first point, I felt nervous about changing it too much without taking a pass over the entire chapter.

Your other comments make sense to me.  Pushing an update to address them in a moment.

Reply:
I think your last commit improved things. I understand the hesitance and a better description would take more work then we're comfortable with.